js/views/aliases-list.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * ownCloud - Mail
+ *
+ * This file is licensed under the Affero General Public License version 3 or
+ * later. See the COPYING file.
+ *
+ * @author Tahaa Karim <tahaalibra@gmail.com>
+ * @copyright Tahaa Karim 2016
+ */
+
+define(function(require) {
+	'use strict';
+
+	var $ = require('jquery');
+	var Marionette = require('marionette');
+	var Handlebars = require('handlebars');
+	var AliasesListTemplate = require('text!templates/aliases-list.html');
+	var AliasesController = require('controller/aliasescontroller');
+
+	return Marionette.CompositeView.extend({
+		template: Handlebars.compile(AliasesListTemplate),
+		templateHelpers: function() {
+			var aliases = this.aliases;
+			return {
+				aliases: aliases
+			};
+		},
+		ui: {
+			'deleteButton' : 'button'
+		},
+		events: {
+			'click @ui.deleteButton': 'deleteAlias'
+		},
+		aliases: null,
+		currentAccount: null,
+		initialize: function(options) {
+			this.currentAccount = options.currentAccount;
+			this.aliases = this.currentAccount.get('aliases');
+		},
+		deleteAlias: function() {
+			event.stopPropagation();
+			var target = $(event.target);
+			target.removeClass('icon-delete').addClass('icon-loading-small');
+			var aliasId = target.val();
+			var _this = this;
+			var deletingAlias = AliasesController.deleteAlias(this.currentAccount, aliasId);
+
+			$.when(deletingAlias).done(function() {
+				var aliases = [];
+				var json = require('state').accounts.get(_this.currentAccount.get('accountId')).toJSON();
+				for (var x in json.aliases) {
+					if(json.aliases[x].id!=aliasId)
+					aliases.push(json.aliases[x]);
+				}
+				require('state').accounts.get(_this.currentAccount.get('accountId')).unset('aliases');
+				require('state').accounts.get(_this.currentAccount.get('accountId')).set({'aliases': aliases});
```

Comment:
why don't you simple use ``_this.currentAccount.get('accountId')``. Isn't it the same object that you'll get with ``require('state').accounts.get(_this.currentAccount.get('accountId'))``. This confuses me